### ISSUE TITLE
ParHeapOr, ParAnd: avoid allocations while using sync.Pool

### ISSUE THREAD

**Issue by maciej** [RoaringBitmap/roaring#167]
[staticcheck](https://staticcheck.io/docs/staticcheck) reports
```
parallel.go:224:12: argument should be pointer-like to avoid allocations (SA6002)
parallel.go:243:12: argument should be pointer-like to avoid allocations (SA6002)
```
docs for this issue can be found here: https://github.com/dominikh/go-tools/blob/master/cmd/staticcheck/docs/checks/SA6002

**Comment by lemire** [comment]
Are we needlessly copying containers?

**Comment by maciej** [comment]
No, `ParOr` (soon to be `ParHeapOr`) and `ParAnd` needlessly allocate container slices. A slice is nothing more than a struct of pointer to array, length and capacity, no more than 3 int fields. It's a minor thing as you can see, however since it was marked by staticcheck (that @dgryski referred to in #165) I've opened a ticket to handle that.
